pandas/core/indexes/datetimes.py

```diff
@@ -811,9 +811,7 @@ def slice_indexer(self, start=None, end=None, step=None, kind=None):
                     end_casted = self._maybe_cast_slice_bound(end, "right", kind)
                     mask = (self <= end_casted) & mask
 
-                # pandas\core\indexes\datetimes.py:764: error: "bool" has no
-                # attribute "nonzero"  [attr-defined]
-                indexer = mask.nonzero()[0][::step]  # type: ignore[attr-defined]
+                indexer = mask.nonzero()[0][::step]
```

Comment:
probably no harm in using cast, if there are guarantees that mask is always an ndarray by the time we get here.

and the change on L805 wouldn't be needed.

Reply:
sure, will update

Reply:
changed my mind on this one and kept it, since i think passing None for both left and right will lead to a non-ndarray `mask`.  updated the other comment

Reply:
sure